lib/WUI/CMakeLists.txt

```diff
@@ -9,16 +9,23 @@ target_sources(
             alsockets.c
             wui_api.c
             http/fs.c
-            http/handler.c
-            http_handler_default.c
-            http_lifetime.c
-            http/httpd.c
-            http/httpd_post.c
-            http/multipart_parser.c
-            http/upload_state.cpp
+            http_lifetime.cpp
+            nhttp/multipart_parser.c
+            nhttp/upload_state.cpp
             ethernetif.c
             sntp/sntp_client.c
             sntp/sntp.c
+            automata/core.cpp
+            nhttp/common_selectors.cpp
+            nhttp/gcode_upload.cpp
+            nhttp/gen_once.cpp
+            nhttp/handler.cpp
+            nhttp/headers.cpp
+            nhttp/req_parser.cpp
+            nhttp/server.cpp
+            nhttp/static_mem.cpp
+            nhttp/status_page.cpp
```

Comment:
In general, I'm in favor of that. Only, I probably don't want to do it right now, because:

* I tend to work in a way that chunk of related functionality develops itself somewhere and then is torn off once it stabilizes a bit. I don't think this is stabilized enough for that.
* This PR is already too big and I'd prefer to get it in sooner than later.
* There currently are some reverse dependencies. Part of them will disappear when refactoring the link between the current http code and the gcode upload code tailored for the previous implementation, part will need a bit of design to do right.

So it kind of is my longer-term plan from the beginning, that this should eventually be separated. But right now I'd prefer to make it work first.

Reply:
sure, no problem. Can be done later :)